### PR TITLE
refactor: changed the way filters are added and rendered

### PR DESCRIPTION
**Pull request by nchietala**

### CODE REVIEW

File: OpenMTRDemo/Forms/TiledFiltersForm.cs [hunk 1]
```diff
@@ -111,9 +111,9 @@ private void Pane_DblClickHandler(object sender, EventArgs e)
                 }
                 expandedImageForm.Close();
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                MessageBox.Show("Error Opening Image", "Error", MessageBoxButtons.OK);
+                MessageBox.Show(ex + "\n" + ex.Message);
```
Comment (cclinard84) [inline OpenMTRDemo/Forms/TiledFiltersForm.cs]: error

Comment (cclinard84) [reply OpenMTRDemo/Forms/TiledFiltersForm.cs]: I don't think the user needs all this information. We possibly need to give more info than previously done but this is too much.

File: OpenMTRDemo/Forms/ExpandedImageForm.cs [hunk 2]
```diff
@@ -42,45 +56,21 @@ public override void loadToolStripMenuItem_Click(object sender, EventArgs e)
...
-                        {
-                            Cv2.Canny(imageToFilter.Clone(), imageToFilter, CannyThreshold1Slider.Value, CannyThreshold2Slider.Value);
-                        }
-                        else
-                        {
-                            Cv2.Sobel(imageToFilter, imageToFilter, MatType.CV_8U, xorder: 1, yorder: 0, ksize: -1);
-                        }
-                        break;
-                }
+                filter.ApplyFilter(image);
```
Comment (cclinard84) [inline OpenMTRDemo/Forms/ExpandedImageForm.cs]: Nice job!

File: OpenMTRDemo/Filters/testfilter.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenMTRDemo.Filters
+{
+    class testfilter
```
Comment (cclinard84) [inline OpenMTRDemo/Filters/testfilter.cs]: Is this class necessary to leave in?

File: OpenMTRDemo/Forms/TiledFiltersForm.cs (hunk 1 above)
Comment (nchietala) [reply OpenMTRDemo/Forms/TiledFiltersForm.cs]: Oh, oops. I forgot I still had that in there. That was just so I could debug, the catch all try block was preventing me from debugging before I changed it.

**Pull request merged**